src/main/java/org/brapi/test/BrAPITestServer/service/core/ListService.java

```diff
@@ -284,12 +285,12 @@ private void updateEntity(ListEntity entity, @Valid ListNewRequest list) throws
 					itemEntity.setPosition(iter.nextIndex());
 					itemEntity.setItem(item);
 					itemEntity.setList(entity);
-					items.add(itemEntity);
+					entity.getData().add(itemEntity);
 				}
 			}
-			entity.setData(items);
 		} else {
-			entity.setData(new ArrayList<>());
```

Comment:
Does this also need a null check on getData() like was added above?

Reply:
The tests all pass as is, but it's probably a good idea. Added it, too.